chain/chain/src/runtime/mod.rs

```diff
@@ -101,10 +102,14 @@ impl NightshadeRuntime {
         let trie_viewer = TrieViewer::new(trie_viewer_state_size_limit, max_gas_burnt_view);
         let flat_storage_manager = FlatStorageManager::new(store.clone());
         let shard_uids: Vec<_> = genesis_config.shard_layout.shard_uids().collect();
+        // TODO(reshardingV3) Recursively calculate resharding parents for `shard_uids`.
```

Comment:
That would require iterating through previous epochs, which is a bit cumbersome, see https://github.com/near/nearcore/blob/3d0fd26d19f9107056935da1ae58d8e0a557dba9/chain/client/src/sync/epoch.rs#L232-L234.
Wanted to gather some feedback before I continue with this approach.